src/Workspaces/Core/Portable/Shared/Extensions/SyntaxGeneratorExtensions_CreateEqualsMethod.cs

```diff
@@ -309,6 +309,36 @@ public static IMethodSymbol CreateEqualsMethod(this Compilation compilation, Imm
             return statements.ToImmutableAndFree();
         }
 
+        private static void AddReferenceNotNullCheck(
+            SyntaxGenerator factory, Compilation compilation, ParseOptions parseOptions, SyntaxNode otherNameExpression, ArrayBuilder<SyntaxNode> expressions)
+        {
+            var nullLiteral = factory.NullLiteralExpression();
+            if (factory.SyntaxGeneratorInternal.SupportsPatterns(parseOptions))
```

Comment:
📝 Need to specifically check further for supporting _not_ patterns, and use `!(obj is null)` for C# 7

Reply:
done.